Assign specific, unique ports for pprof (Agent, Operator, Hubble Relay)

christarazi opened the pull request:
This allows users to enable pprof for both cilium-agent and
cilium-operator on the same machine without running into:

```
level=warning msg="Unable to serve pprof API" error="listen tcp
127.0.0.1:6060: bind: address already in use" subsys=pprof
```

It also provides users with the ability to specify the port via a flag.
Note that this flag is not exposed via Helm because it is a flag mainly
used for debugging, and shouldn't need to be exposed fully to the user.
A developer / advanced user can just as easily edit the daemonset or the
deployment object to append an extra arg. Also, given the fact that the
ports are unique across the components, that reduces the need to
provide a knob via Helm.

To avoid port collisions, the Agent, Operator, and Hubble Relay are all
assigned the following specific ports respectively: 6060, 6061, 6062.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

christarazi committed:
hubble, daemon, operator, option, pprof: Assign specific default port

This allows users to enable pprof for both cilium-agent and
cilium-operator on the same machine without running into:

```
level=warning msg="Unable to serve pprof API" error="listen tcp
127.0.0.1:6060: bind: address already in use" subsys=pprof
```

It also provides users with the ability to specify the port via a flag.
Note that this flag is not exposed via Helm because it is a flag mainly
used for debugging, and shouldn't need to be exposed fully to the user.
A developer / advanced user can just as easily edit the daemonset or the
deployment object to append an extra arg. Also, given the fact that the
ports are unique across the components, that reduces the need to
provide a knob via Helm.

To avoid port collisions, the Agent, Operator, and Hubble Relay are all
assigned the following specific ports respectively: 6060, 6061, 6062.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

aditighag reviewed (approved):
>It's worth mentioning that both the cilium-agent and cilium-operator
share the Cilium configmap so if a port was provided, then it will be
used for both, and will cause the same port collision issue. The user
must edit the cilium-operator deployment or the cilium-agent daemonset
directly in order to pass different ports to each component to properly
workaround the port collision.

Why not use different default ports?

christarazi committed:
docs: Update system requirements with new pprof ports

Following the commit to (re)assign the pprof default port, update the
documentation with the new information.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

christarazi commented on operator/option/config.go:
```diff
@@ -76,6 +76,9 @@ const (
 	PProf = "pprof"
 
+	// PProfPort is the port that the pprof listens on
+	PProfPort = "operator-pprof-port"
```
This doesn't need to be called out specifically for operator. This is leftover from when I was considering Helm changes. I'll change this back to `pprof-port`.

twpayne reviewed (approved):
👍

kaworu reviewed (approved):
Thanks @aditighag!
